Update README, precommits and other configs with some cosmetic changes (#21)
predictionmachine/pm-coding-template: docs/coding-standard.md

# Coding Standard

> Documents and Illustrates Coding Standards used at Prediction Machine.

Reviewer: ```suggestion
Documents and Illustrates Coding Standards used at [Prediction Machine](https://predmachine.com).
```

As you ramp up on contributing code to Prediction Machine,
please review the basics:

## Dev Environment

* Mac/Linux preferred. For Windows, use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
* do use separate [virtual env](https://docs.python.org/3/library/venv.html) for each repo
* IDE:
  * For long-term contributors we prefer that you use the
  [PyCharm](https://www.jetbrains.com/pycharm/) IDE (if you are already
  using IntelliJ, IntelliJ with Python plugin may work)
  * [VSCode](https://code.visualstudio.com/) may also be viable if that's your strong preference
* Pair Programming:
  * (beta) We are evaluating [Code With Me](https://plugins.jetbrains.com/plugin/14896-code-with-me)
as a way to pair on code, a surprisingly fruitful technique
* Dependencies:
  * Target `python3.8` for compatibility (eg with AWS Lambda)
  * Pin dependency versions in requirements.txt with `~=`, eg `flake8~=3.8.4`
* file names are normally all lowercase

> * when file names must have multiple words, use _ not -

Reviewer: ```suggestion
* when file names must have multiple words, use "_", not "-"
```

> * Organize code in namespace and files. `tests` dir inside the namespace

Reviewer: ```suggestion
* Organize code in namespace and files. `tests` dir at the top level
```
